packages/eas-cli/src/commands/update/configure.ts

```diff
@@ -94,25 +191,74 @@ export default class UpdateConfigure extends EasCommand {
+    const [androidWorkflow, iosWorkflow] = await Promise.all([
+      resolveWorkflowAsync(projectDir, Platform.ANDROID),
+      resolveWorkflowAsync(projectDir, Platform.IOS),
+    ]);
 
-    await configureProjectForEASUpdateAsync(projectDir, exp, isBare);
+    const defaultRuntimeVersions = {
+      android:
+        androidWorkflow === Workflow.GENERIC
+          ? DEFAULT_BARE_RUNTIME_VERSION
+          : DEFAULT_MANAGED_RUNTIME_VERSION,
+      ios:
+        iosWorkflow === Workflow.GENERIC
+          ? DEFAULT_BARE_RUNTIME_VERSION
+          : DEFAULT_MANAGED_RUNTIME_VERSION,
+    };
 
-    Log.addNewLineIfNone();
-    if (isBare) {
-      Log.log(
-        `🧐 It seems you are on the bare workflow! Please also update your native files. You can do this by either running ${chalk.bold(
-          'eas build:configure'
-        )} or manually editing Expo.plist/AndroidManifest.xml. ${learnMore(
-          'https://expo.fyi/eas-update-config.md#native-configuration'
-        )}`
-      );
-    } else {
-      Log.log(`🎉 Your app is configured to run EAS Update!`);
+    let updatedExp: ExpoConfig;
+    try {
+      updatedExp = await configureAppJSONForEASUpdateAsync({
+        projectDir,
+        exp,
+        platform,
+        defaultRuntimeVersions,
+      });
+    } catch (e) {
+      if (androidWorkflow === Workflow.GENERIC || iosWorkflow === Workflow.GENERIC) {
+        Log.warn(
+          `You will also have to manually edit Expo.plist/AndroidManifest.xml. ${learnMore(
+            'https://expo.fyi/eas-update-config.md#native-configuration'
+          )}`
+        );
+      }
+      Log.addNewLineIfNone();
+      throw e;
     }
+
+    // configure native files for EAS Update
+    const nativeFilesToSync = [];
+    if (
+      [RequestedPlatform.Android, RequestedPlatform.All].includes(platform) &&
+      androidWorkflow === Workflow.GENERIC
+    ) {
+      nativeFilesToSync.push(syncAndroidUpdatesConfigurationAsync(projectDir, updatedExp));
+      Log.withTick('Configured AndroidManifest.xml for EAS Update');
+    }
+    if (
+      [RequestedPlatform.Ios, RequestedPlatform.All].includes(platform) &&
+      iosWorkflow === Workflow.GENERIC
+    ) {
+      nativeFilesToSync.push(syncIosUpdatesConfigurationAsync(projectDir, updatedExp));
+      Log.withTick('Configured Expo.plist for EAS Update');
+    }
+    await Promise.all(nativeFilesToSync);
+
+    Log.addNewLineIfNone();
+    Log.log(`🎉 Your app is configured to run EAS Update!`);
+  }
+}
+
+function isRuntimeEqual(
+  runtimeVersionA: string | { policy: 'sdkVersion' | 'nativeVersion' },
+  runtimeVersionB: string | { policy: 'sdkVersion' | 'nativeVersion' }
+): boolean {
+  if (typeof runtimeVersionA === 'string' && typeof runtimeVersionB === 'string') {
+    return runtimeVersionA === runtimeVersionB;
+  }
+  if (typeof runtimeVersionA === 'object' && typeof runtimeVersionB === 'object') {
+    return runtimeVersionA.policy === runtimeVersionB.policy;
   }
+  return false;
```

Comment:
```suggestion
  if (typeof runtimeVersionA === 'string' && typeof runtimeVersionB === 'string') {
    return runtimeVersionA === runtimeVersionB;
  } else if (typeof runtimeVersionA === 'object' && typeof runtimeVersionB === 'object') {
    return runtimeVersionA.policy === runtimeVersionB.policy;
  } else {
    return false;
  }
```

IMO this is more readable.